### DISCUSSION TITLE
Multiple Batteries, switching to float

### DISCUSSION THREAD

**Discussion by maxx8888** [Louisvdw/dbus-serialbattery#787] [General]
Hi

Setup:
3x 280AH 18cell EVE Lifepo Packs, connected via RS485 and Gigavolt Aggregate Batteries.
Aggregate Batteries Controlling disabled, as the Balance Handling is not good there (if Balance is used, it will stress the Batteries the whole day, as it doesn't lower Voltage after reaching a balanced state)

Serial Battery:
MAX_CELL_VOLTAGE = 3.450
FLOAT_CELL_VOLTAGE  = 3.340
LINEAR_LIMITATION_ENABLE = TRUE
CVCM_ENABLE = TRUE
CELL_VOLTAGE_DIFF_KEEP_MAX_VOLTAGE_UNTIL = 0.004
CELL_VOLTAGE_DIFF_TO_RESET_VOLTAGE_LIMIT = 0.040

All 3 Battery Packs are charged up to CVL 3.45V. As soon as the first one reaches Diff lower than 0.004, CVL is reduced to 3.34V.
The other 2 Batteries in the Pack doesn't have time to Balance at all and get unbalanced more day by day. After a few days, the additional Batteries have cells so much off that system starts bouncing.

First Battery reaches 0.004 -> lower Voltage
During lowering Voltage -> Battery 2 or 3 report diff higher than 0.040 -> increase Voltage

For my opinion it would make absolutly sense to keep Balance Voltage until all cells in all Packs are reaching a difference of "CELL_VOLTAGE_DIFF_KEEP_MAX_VOLTAGE_UNTIL". Then lower voltage.
Otherwise, always one Pack will drift away with some cell voltages.

Does somebody else face similar Problems?
Any idea how i could resolve this Issue by Parameter Tuning?

I've also read a releated discussion here:
https://github.com/Louisvdw/dbus-serialbattery/issues/659

**Comment by wetty** [comment]
Hello,

I have the same setup, but only with two times 280AH 18cell EVE Lifepo Packs, connected via RS485 and Gigavolt Aggregate Batteries.

And I have the same problem. I try to minimize it with setting the better balanced pack to a balance trigger voltage in the JKBMS Settings of 0.005 and the other to 0.003. 
CELL_VOLTAGE_DIFF_KEEP_MAX_VOLTAGE_UNTIL = 0.003

It is still not a perfect solution. Maybe I write a Node-Red flow that disables balancing on the better pack until the other pack shows the same call difference and then enable balancing on the first pack again-

**Comment by mr-manuel** [comment]
Did you open a discussion/issue in Gigavolt Aggregate Batteries repository? The aggregator should handle this. This driver cannot do anything.

**Comment by wetty** [comment]
I dont think it is a Aggregate Batteries problem, as Serialbattery is in my case doing all the balancing.
So Serialbattery is switching from Load Voltage to Float Voltage, when only one of the battery packs is below or equal of the cell difference in the serialbattery.config
If the second pack has a cell difference from 0.06 V  and the first from 0.003 it is changing to the float voltage.
So far as I know/think does Aggregate Batteries not deliver a Delta Cell Voltage

**Reply by mr-manuel** [reply]
The answer is hidden in your post :-)

> So Serialbattery is switching from Load Voltage to Float Voltage, when only one of the battery packs

dbus-serialbattery only handles one battery, the aggregator is aggregating all batteries. Therefore the aggregator should be the master BMS selected in the DVCC settings (Venus OS `>= 3.0.0`) and only switch from absoprtion voltage to float voltage, if all batteries switched to float. It doesn't matter, if a single battery switches to float, since only one master BMS can be selected that is controlling the voltage and current.

Therefore you need the aggregator, since it should respect the need of all the batteries and tell the charger/inverter what to do.

**Reply by sruchard1** [reply]
@mr-manuel i have a problem not exactly with the agregator. i have 6 felicity 15kwh battery pack and i have used the dbserialbattery to read the bms data down to cells voltage as well as publish on venus os large. but if the battery packs are linkes together with the rs485 for synch communication between themselves, the dbserialbatteries will stop working and all the initially published data will disapper. can you enlighten me on how to get the total pack data when they are all linked with their data cable to the master battry. thank you

**Comment by wetty** [comment]
But the float voltage is the one defined in serialbattery. And SerialBattery is switching to float when minimum call difference is reached and hold for the defined amount of time.
The Aggregator does not show any cell difference and venus os does switch to float based on maximum voltage not on the defference.

**Reply by mr-manuel** [reply]
> But the float voltage is the one defined in serialbattery. And SerialBattery is switching to float when minimum call difference is reached and hold for the defined amount of time.

Yes, but only for ONE battery seperately. The battery can request what it want, when the master/controlling BMS (which should be the aggregator) is the controlling one. Only the controlling BMS can change voltage and current for all. Since you have all batteries connected in parallel there is no way to set voltage and current per battery.

![grafik](https://github.com/Louisvdw/dbus-serialbattery/assets/12592765/0834df07-413d-444c-ace3-75b4803dc24b)

It's like if you have 5 children:
* 1 wan't to go in the park
* 2 want to play basket
* 2 want to go swim

But you, the father, wants to go to the football stadion therefore all go to the football stadion. The master decides what it's decided.

> The Aggregator does not show any cell difference and venus os does switch to float based on maximum voltage not on the defference.

Venus OS switch the voltage based on what the master/controlling BMS say. The aggregator do not have to know any cell voltage differences, it only has to switch to float, if all batteries switched to float. If 2 batteries are in float and oane in absorption the battery aggregator has to stay in absorption until all batteries switched to float.

**Comment by wetty** [comment]
Just to understand the functionality right.

Current Behaviour:
Aggregate gets the info from in my case Serialbattery1, that min cell difference is reached for the amount of time.
Aggregate tells Venus OS to switch to the float Voltage from serialbattery.config?

Wanted behaviour:
Aggregate gets the info from in my case Serialbattery1, that min cell difference is reached for the amount of time.
Aggregate wait until Serialbattery2 has also reached the conditions.
Then Aggregate tells Venus OS to switch to the float Voltage from serialbattery.config?

Here are some screenshot from my installation. On Battery in float the other not.

![Bildschirmfoto 2023-09-25 um 16 58 52](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/06c2c10f-c151-4c25-be4a-81c8b15d5ce6)
![Bildschirmfoto 2023-09-25 um 16 58 24](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/aae64ba4-39c1-469b-8522-d501fad40119)
![Bildschirmfoto 2023-09-25 um 16 58 41](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/6b1ab217-ce07-45b1-81ae-56a6f236e353)

**Reply by mr-manuel** [reply]
What does the AggregateBatteries say? That screenshot is missing.

Do you have at least Venus OS v3.00 installed? If not you have to upgrade. After post a screenshot of what BMS you have selected as controlling BMS.

Prior Venus OS v3.00 the controlling BMS is selected sporadically since it get's the first responding BMS.

**Comment by wetty** [comment]
Latest Venus Release Candidate
![Bildschirmfoto 2023-09-25 um 17 20 39](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/ffc92e1f-2bf2-4dd6-9001-2dfce915e6bd)
![Bildschirmfoto 2023-09-25 um 17 21 28](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/928541a3-ed23-4175-833e-b9a4904e4c13)

**Reply by mr-manuel** [reply]
Why you don't post the same parameters page as for the other? Please do that.
Why have you enabled SVS, SCS, STS?

**Comment by wetty** [comment]
If I dont enabled i.e. SVS then I dont reach my 62.1V Charge Voltage, as my Victron seems to measure a higher Voltage then my BMSse. And I have double check the voltage with two different Voltmeters  :-)

Sorry here the Parameters from Aggregate

![Bildschirmfoto 2023-09-25 um 17 26 56](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/b9a49a56-1db1-44ed-bf3f-08d1280542ca)

**Reply by mr-manuel** [reply]
Here you see that the AggregateBatteries goes to float, if the first battery goes to float. In my opinion that's wrong.

You can configure a voltage drop in the dbus-serialbattery. Check the config.default.ini for that. What is the voltage drop you measured between battery and Victron?

If you enable STS, SVS and SCS it won't work reliable from what I experienced.

**Reply by wetty** [reply]
See the solved Issue Multiple Batteries and switching to float https://github.com/Dr-Gigavolt/dbus-aggregate-batteries/issues/52
with the following answer: 
Otherwise (OWN_CHARGE_PARAMETERS = False) my code cannot know why one of your Serial Battery instances reduces the CVL. I must assume the worst case (danger of over-charging of some cell(s)) and therefore the minimum of all CVLs is passed to the ESS.

**Reply by mr-manuel** [reply]
Thats not all true. `dbus-serialbattery` provides planty of information. Let's see where this issue lead to. In case it's not leading into the direction you want you can use `dbus-mqtt-battery` from my repo and aggregate the battery data yourself. This allows you also to decide when to lower the CVL and when not.

**Comment by wetty** [comment]
Okay,
did try the voltage drop several month/weeks ago but it does not work. I will see to test it again. And even try to measure it.
With SVS on its working okay since 10 month now

**Comment by wetty** [comment]
Screenshots from the difference just after disable svs

![Bildschirmfoto 2023-09-25 um 17 51 07](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/401fd1d9-e9d6-4beb-a1ba-3c859b10ccfa)
![Bildschirmfoto 2023-09-25 um 17 51 23](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/fe295e01-fc74-4b1e-9967-62c0ee39b07a)

**Comment by wetty** [comment]
I have opened a new discussion to explain my Voltage Drop Problem
What does VOLTAGE_DROP exactly? #822
